garden/src/hooks/useSearchResults.ts

```diff
@@ -120,13 +126,29 @@ export const useSearchResults = (): SearchResultsState => {
     [updateSearchParams],
   );
 
+  const facets = useMemo(() => {
+    if (!searchResult?.facets) return [];
+    return Object.entries(searchResult.facets).map(
+      ([name, values]: [string, Record<string, number>]) => ({
+        name,
+        values: Object.entries(values)
+          .map(([value, count]: [string, number]) => ({ value, count }))
+          .sort((a, b) => {
+            return (
+              2 * (b.count - a.count) - (Number(selectedFilters[name]?.includes(a.value)) || -1)
```

Comment:
This is a little hard to follow. I don't understand why we're multiplying by 2 and being inside this nested map and sort makes it harder. Could you break this up and/or comment it a bit to convey what's going on?

Reply:
Yes this was indeed a bit strange. Basically this is to ensure that _selected_ filters appear at the top of the filters list. But I also want filters with a high number of matching results to appear near the top. I'll  try to write this in a way that's more clear